src/EditorFeatures/Core/ValueTracking/ValueTrackingService.Visitor.cs

```diff
@@ -0,0 +1,232 @@
+            {
+                await AddOperationAsync(invocationOperation, invocationOperation.TargetMethod, cancellationToken).ConfigureAwait(false);
+                await TrackArgumentsAsync(invocationOperation.Arguments, cancellationToken).ConfigureAwait(false);
+            }
+
+            private Task VisitReferenceAsync(IOperation operation, CancellationToken cancellationToken)
+            {
+                Debug.Assert(operation is
+                    ILocalReferenceOperation or
+                    IParameterReferenceOperation or
+                    IFieldReferenceOperation or
+                    IPropertyReferenceOperation);
+
+                if (IsArgument(operation, out var argumentOperation) && argumentOperation.Parameter is not null)
+                {
+                    if (argumentOperation.Parameter.IsRefOrOut())
+                    {
+                        // Always add ref or out parameters to track as assignments since the values count as 
+                        // assignments across method calls for the purposes of value tracking.
+                        return AddOperationAsync(operation, argumentOperation.Parameter, cancellationToken);
+                    }
+
+                    // If the parameter is not a ref or out param, track the reference assignments that count
+                    // as input to the argument being passed to the method.
+                    return AddReference(operation, cancellationToken);
+                }
+
+                if (IsReturn(operation))
+                {
+                    // If the reference is part of a return operation we want to track where the values come from
+                    // since they contribute to the "output" of the method and are relavent for value tracking.
+                    return AddReference(operation, cancellationToken);
+                }
+
+                return Task.CompletedTask;
+
+                Task AddReference(IOperation operation, CancellationToken cancellationToken)
+                    => operation switch
+                    {
+                        IParameterReferenceOperation parameterReference => AddOperationAsync(operation, parameterReference.Parameter, cancellationToken),
+                        IFieldReferenceOperation fieldReferenceOperation => AddOperationAsync(operation, fieldReferenceOperation.Member, cancellationToken),
+                        IPropertyReferenceOperation propertyReferenceOperation => AddOperationAsync(operation, propertyReferenceOperation.Member, cancellationToken),
+                        ILocalReferenceOperation localReferenceOperation => AddOperationAsync(operation, localReferenceOperation.Local, cancellationToken),
+                        _ => Task.CompletedTask
+                    };
+            }
+
+            private static bool IsArgument(IOperation? operation, [NotNullWhen(returnValue: true)] out IArgumentOperation? argumentOperation)
```

Comment:
seems like htis is more 'IsContainedInArgument' since this walks arbitrary far up.  i would have expected IsArgument to just be doing a direct check of the node itselr (or maybe just the parent).

Reply:
Changed to `IsContainedIn<T>` to use generically across a few things